core/rfb.js

```diff
@@ -398,11 +398,22 @@ export default class RFB extends EventTargetMixin {
     _connect() {
         Log.Debug(">> RFB.connect");
 
-        Log.Info("connecting to " + this._url);
-
         try {
-            // WebSocket.onopen transitions to the RFB init states
-            this._sock.open(this._url, this._wsProtocols);
+            // WebChannel.onopen transitions to the RFB init states
+            if (typeof this._urlOrChannel === "string") {
+                Log.Info(`connecting to ${this._urlOrChannel}`);
+                this._sock.open({ uri: this._urlOrChannel, protocols: this._wsProtocols });
+            } else {
+                let proto = Object.getPrototypeOf(this._urlOrChannel);
+                let protoName = proto && proto.constructor && proto.constructor.name || "";
```

Comment:
I would prefer if we can duck type this rather than only supporting those low level objects.